### DISCUSSION TITLE
What are the 'module' and 'source' top-level fields in 'package.json' used for?

### DISCUSSION THREAD

**Discussion by AsteriskZuo** [callstack/react-native-builder-bob#323] [Q&A]
{
  "name": "@sendbird/uikit-react-native",
  "version": "2.0.0",
  "description": "react-native-uikit",
  "main": "lib/commonjs/index",
  "module": "lib/module/index",
  "types": "lib/typescript/src/index.d.ts",
  "react-native": "src/index",
  "source": "src/index",
  ...
}

**Comment by satya164** [comment]
`module` field is used by Webpack and other modern bundlers when building for Web - essentially its code using import/export syntax.
`source` is used for development in the project, we read this field in babel config to be able to alias the library for example

**Reply by AsteriskZuo** [reply]
Are these two fields important? After I delete these two fields, the project compile and run, did not receive a little impact?

**Reply by satya164** [reply]
Removing `source` will break development workflow.

`module` is for consumers so that tree shaking can work in supported bundlers. If you remove then the package will still work but tree shaking won't (when applicable).

All those fields are there to support various cases. So I suggest not removing them without a good reason.

**Reply by AsteriskZuo** [reply]
Thank you very much.
